### ISSUE TITLE
Calling .Version() on a resource throws an error

### ISSUE THREAD

**Issue by aaron-collier** [sul-dlss-labs/taco#394]
When I try to compare the Version on a resource for update, I'm getting the following error on a newResource from parameters:

```
{"message":"interface conversion: interface {} is json.Number, not int","stack":"goroutine 27 [running]:
runtime/debug.Stack(0xc4204c2d20, 0x1f4, 0xc420405080)
	/usr/local/Cellar/go/1.9.3/libexec/src/runtime/debug/stack.go:24 +0xa7
github.com/sul-dlss-labs/taco/middleware.(*Recovery).ServeHTTP.func1(0x4c5f5e0, 0xc4204c2d20, 0xc42052f620)
	/Users/sul.amcollie/go/src/github.com/sul-dlss-labs/taco/middleware/recovery.go:38 +0x91
panic(0x471a3e0, 0xc420405080)
	/usr/local/Cellar/go/1.9.3/libexec/src/runtime/panic.go:491 +0x283
github.com/sul-dlss-labs/taco/datautils.(*JSONObject).GetI(0xc42000e1f0, 0x47ee33a, 0x7, 0x0)
	/Users/sul.amcollie/go/src/github.com/sul-dlss-labs/taco/datautils/jsonobject.go:17 +0xb8
github.com/sul-dlss-labs/taco/datautils.(*Resource).Version(0xc42000e1f0, 0xc420270403)
	/Users/sul.amcollie/go/src/github.com/sul-dlss-labs/taco/datautils/resource.go:60 +0x42
github.com/sul-dlss-labs/taco/handlers.(*updateResourceEntry).Handle(0xc420549280, 0xc42046c700, 0xc420270403, 0xb, 0x4709e00, 0xc420511440, 0x0, 0x4c64a23)
	/Users/sul.amcollie/go/src/github.com/sul-dlss-labs/taco/handlers/update_resource.go:46 +0x34d
github.com/sul-dlss-labs/taco/generated/restapi/operations.(*UpdateResource).ServeHTTP(0xc4204d1a00, 0x4c5ea60, 0xc42052f720, 0xc42046c700)
	/Users/sul.amcollie/go/src/github.com/sul-dlss-labs/taco/generated/restapi/operations/update_resource.go:56 +0x229
github.com/sul-dlss-labs/taco/vendor/github.com/go-openapi/runtime/middleware.NewOperationExecutor.func1(0x4c5ea60, 0xc42052f720, 0xc42046c700)
	/Users/sul.amcollie/go/src/github.com/sul-dlss-labs/taco/vendor/github.com/go-openapi/runtime/middleware/operation.go:28 +0x6e
net/http.HandlerFunc.ServeHTTP(0xc420325af0, 0x4c5ea60, 0xc42052f720, 0xc42046c700)
	/usr/local/Cellar/go/1.9.3/libexec/src/net/http/server.go:1918 +0x44
github.com/sul-dlss-labs/taco/vendor/github.com/go-openapi/runtime/middleware.NewRouter.func1(0x4c5ea60, 0xc42052f720, 0xc42046c600)
	/Users/sul.amcollie/go/src/github.com/sul-dlss-labs/taco/vendor/github.com/go-openapi/runtime/middleware/router.go:74 +0x364
net/http.HandlerFunc.ServeHTTP(0xc42052efe0, 0x4c5ea60, 0xc42052f720, 0xc42046c600)
	/usr/local/Cellar/go/1.9.3/libexec/src/net/http/server.go:1918 +0x44
github.com/sul-dlss-labs/taco/vendor/github.com/go-openapi/runtime/middleware.Redoc.func1(0x4c5ea60, 0xc42052f720, 0xc42046c600)
	/Users/sul.amcollie/go/src/github.com/sul-dlss-labs/taco/vendor/github.com/go-openapi/runtime/middleware/redoc.go:72 +0x94
net/http.HandlerFunc.ServeHTTP(0xc420404800, 0x4c5ea60, 0xc42052f720, 0xc42046c600)
	/usr/local/Cellar/go/1.9.3/libexec/src/net/http/server.go:1918 +0x44
github.com/sul-dlss-labs/taco/vendor/github.com/go-openapi/runtime/middleware.Spec.func1(0x4c5ea60, 0xc42052f720, 0xc42046c600)
	/Users/sul.amcollie/go/src/github.com/sul-dlss-labs/taco/vendor/github.com/go-openapi/runtime/middleware/spec.go:45 +0x73
net/http.HandlerFunc.ServeHTTP(0xc420404840, 0x4c5ea60, 0xc42052f720, 0xc42046c600)
	/usr/local/Cellar/go/1.9.3/libexec/src/net/http/server.go:1918 +0x44
github.com/sul-dlss-labs/taco/middleware.(*RequestLogger).ServeHTTP(0xc42036e8c0, 0x4c5f5e0, 0xc4204c2d20, 0xc42046c600)
	/Users/sul.amcollie/go/src/github.com/sul-dlss-labs/taco/middleware/logger.go:32 +0xb9
github.com/sul-dlss-labs/taco/middleware.(*Recovery).ServeHTTP(0xc42052f620, 0x4c5f5e0, 0xc4204c2d20, 0xc42046c600)
	/Users/sul.amcollie/go/src/github.com/sul-dlss-labs/taco/middleware/recovery.go:49 +0x90
github.com/sul-dlss-labs/taco/vendor/github.com/honeybadger-io/honeybadger-go.(*Client).Handler.func1(0x4c5f5e0, 0xc4204c2d20, 0xc42046c600)
	/Users/sul.amcollie/go/src/github.com/sul-dlss-labs/taco/vendor/github.com/honeybadger-io/honeybadger-go/client.go:99 +0x94
net/http.HandlerFunc.ServeHTTP(0xc42052f660, 0x4c5f5e0, 0xc4204c2d20, 0xc42046c600)
	/usr/local/Cellar/go/1.9.3/libexec/src/net/http/server.go:1918 +0x44
net/http.serverHandler.ServeHTTP(0xc4205af040, 0x4c5f5e0, 0xc4204c2d20, 0xc42046c600)
	/usr/local/Cellar/go/1.9.3/libexec/src/net/http/server.go:2619 +0xb4
net/http.(*conn).serve(0xc4201d23c0, 0x4c60020, 0xc420404900)
	/usr/local/Cellar/go/1.9.3/libexec/src/net/http/server.go:1801 +0x71d
created by net/http.(*Server).Serve
	/usr/local/Cellar/go/1.9.3/libexec/src/net/http/server.go:2720 +0x288
```

And then I get this error on the existingResource:

```
{"message":"interface conversion: interface {} is float64, not int","stack":"goroutine 11 [running]:
runtime/debug.Stack(0xc42021cd20, 0x1f4, 0xc420298200)
	/usr/local/Cellar/go/1.9.3/libexec/src/runtime/debug/stack.go:24 +0xa7
github.com/sul-dlss-labs/taco/middleware.(*Recovery).ServeHTTP.func1(0x4c5f5e0, 0xc42021cd20, 0xc420200080)
	/Users/sul.amcollie/go/src/github.com/sul-dlss-labs/taco/middleware/recovery.go:38 +0x91
panic(0x471a3e0, 0xc420298200)
	/usr/local/Cellar/go/1.9.3/libexec/src/runtime/panic.go:491 +0x283
github.com/sul-dlss-labs/taco/datautils.(*JSONObject).GetI(0xc420382128, 0x47ee33a, 0x7, 0x0)
	/Users/sul.amcollie/go/src/github.com/sul-dlss-labs/taco/datautils/jsonobject.go:17 +0xb8
github.com/sul-dlss-labs/taco/datautils.(*Resource).Version(0xc420382128, 0xc4202c8013)
	/Users/sul.amcollie/go/src/github.com/sul-dlss-labs/taco/datautils/resource.go:60 +0x42
github.com/sul-dlss-labs/taco/handlers.(*updateResourceEntry).Handle(0xc4205a2e40, 0xc42050c200, 0xc4202c8013, 0xb, 0x4709e00, 0xc4206122d0, 0x0, 0x4c64a23)
	/Users/sul.amcollie/go/src/github.com/sul-dlss-labs/taco/handlers/update_resource.go:46 +0x348
github.com/sul-dlss-labs/taco/generated/restapi/operations.(*UpdateResource).ServeHTTP(0xc4204585e0, 0x4c5ea60, 0xc420634060, 0xc42050c200)
	/Users/sul.amcollie/go/src/github.com/sul-dlss-labs/taco/generated/restapi/operations/update_resource.go:56 +0x229
github.com/sul-dlss-labs/taco/vendor/github.com/go-openapi/runtime/middleware.NewOperationExecutor.func1(0x4c5ea60, 0xc420634060, 0xc42050c200)
	/Users/sul.amcollie/go/src/github.com/sul-dlss-labs/taco/vendor/github.com/go-openapi/runtime/middleware/operation.go:28 +0x6e
net/http.HandlerFunc.ServeHTTP(0xc42029ad20, 0x4c5ea60, 0xc420634060, 0xc42050c200)
	/usr/local/Cellar/go/1.9.3/libexec/src/net/http/server.go:1918 +0x44
github.com/sul-dlss-labs/taco/vendor/github.com/go-openapi/runtime/middleware.NewRouter.func1(0x4c5ea60, 0xc420634060, 0xc42050c000)
	/Users/sul.amcollie/go/src/github.com/sul-dlss-labs/taco/vendor/github.com/go-openapi/runtime/middleware/router.go:74 +0x364
net/http.HandlerFunc.ServeHTTP(0xc4204e1be0, 0x4c5ea60, 0xc420634060, 0xc42050c000)
	/usr/local/Cellar/go/1.9.3/libexec/src/net/http/server.go:1918 +0x44
github.com/sul-dlss-labs/taco/vendor/github.com/go-openapi/runtime/middleware.Redoc.func1(0x4c5ea60, 0xc420634060, 0xc42050c000)
	/Users/sul.amcollie/go/src/github.com/sul-dlss-labs/taco/vendor/github.com/go-openapi/runtime/middleware/redoc.go:72 +0x94
net/http.HandlerFunc.ServeHTTP(0xc420366580, 0x4c5ea60, 0xc420634060, 0xc42050c000)
	/usr/local/Cellar/go/1.9.3/libexec/src/net/http/server.go:1918 +0x44
github.com/sul-dlss-labs/taco/vendor/github.com/go-openapi/runtime/middleware.Spec.func1(0x4c5ea60, 0xc420634060, 0xc42050c000)
	/Users/sul.amcollie/go/src/github.com/sul-dlss-labs/taco/vendor/github.com/go-openapi/runtime/middleware/spec.go:45 +0x73
net/http.HandlerFunc.ServeHTTP(0xc4203665c0, 0x4c5ea60, 0xc420634060, 0xc42050c000)
	/usr/local/Cellar/go/1.9.3/libexec/src/net/http/server.go:1918 +0x44
github.com/sul-dlss-labs/taco/middleware.(*RequestLogger).ServeHTTP(0xc420324330, 0x4c5f5e0, 0xc42021cd20, 0xc42050c000)
	/Users/sul.amcollie/go/src/github.com/sul-dlss-labs/taco/middleware/logger.go:32 +0xb9
github.com/sul-dlss-labs/taco/middleware.(*Recovery).ServeHTTP(0xc420200080, 0x4c5f5e0, 0xc42021cd20, 0xc42050c000)
	/Users/sul.amcollie/go/src/github.com/sul-dlss-labs/taco/middleware/recovery.go:49 +0x90
github.com/sul-dlss-labs/taco/vendor/github.com/honeybadger-io/honeybadger-go.(*Client).Handler.func1(0x4c5f5e0, 0xc42021cd20, 0xc42050c000)
	/Users/sul.amcollie/go/src/github.com/sul-dlss-labs/taco/vendor/github.com/honeybadger-io/honeybadger-go/client.go:99 +0x94
net/http.HandlerFunc.ServeHTTP(0xc4202000a0, 0x4c5f5e0, 0xc42021cd20, 0xc42050c000)
	/usr/local/Cellar/go/1.9.3/libexec/src/net/http/server.go:1918 +0x44
net/http.serverHandler.ServeHTTP(0xc42071e680, 0x4c5f5e0, 0xc42021cd20, 0xc42050c000)
	/usr/local/Cellar/go/1.9.3/libexec/src/net/http/server.go:2619 +0xb4
net/http.(*conn).serve(0xc4201d0000, 0x4c60020, 0xc4205a2000)
	/usr/local/Cellar/go/1.9.3/libexec/src/net/http/server.go:1801 +0x71d
created by net/http.(*Server).Serve
	/usr/local/Cellar/go/1.9.3/libexec/src/net/http/server.go:2720 +0x288
```

So definitely some inconsistency in what is available for the version.

**Comment by jcoyne** [comment]
I think I've addressed this in #374 
https://github.com/sul-dlss-labs/taco/pull/374/files#diff-b59c3123b50baba682c9babe1d65b958R78
and here https://github.com/sul-dlss-labs/taco/pull/374/files#diff-e456c0f4ffea16e9c6622d3d47e8ba18R54

**Comment by aaron-collier** [comment]
Cool @jcoyne - once 374 is in, I can remove the fix I have in 395 :-)
